x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -333,6 +343,68 @@ function restructureConnectorsThatSupportIncident(
   };
 }
 
+function convertNullToUndefined(attribute: SavedObjectAttribute) {
+  return attribute != null ? attribute : undefined;
+}
+
+function removeNullsFromSecurityRules(
+  doc: SavedObjectUnsanitizedDoc<RawAlert>
+): SavedObjectUnsanitizedDoc<RawAlert> {
+  const {
+    attributes: { params },
+  } = doc;
+  return {
+    ...doc,
+    attributes: {
+      ...doc.attributes,
+      params: {
+        ...params,
+        buildingBlockType: convertNullToUndefined(params.buildingBlockType),
+        note: convertNullToUndefined(params.note),
+        index: convertNullToUndefined(params.index),
+        language: convertNullToUndefined(params.language),
+        license: convertNullToUndefined(params.license),
+        outputIndex: convertNullToUndefined(params.outputIndex),
+        savedId: convertNullToUndefined(params.savedId),
+        timelineId: convertNullToUndefined(params.timelineId),
+        timelineTitle: convertNullToUndefined(params.timelineTitle),
+        meta: convertNullToUndefined(params.meta),
+        query: convertNullToUndefined(params.query),
+        filters: convertNullToUndefined(params.filters),
+        riskScoreMapping: params.riskScoreMapping != null ? params.riskScoreMapping : [],
+        ruleNameOverride: convertNullToUndefined(params.ruleNameOverride),
+        severityMapping: params.severityMapping != null ? params.severityMapping : [],
+        threat: params.threat != null ? params.threat : [],
+        threshold:
+          params.threshold != null &&
+          typeof params.threshold === 'object' &&
+          !Array.isArray(params.threshold)
+            ? {
+                field: Array.isArray(params.threshold.field)
+                  ? params.threshold.field
+                  : params.threshold.field === '' || params.threshold.field == null
+                  ? []
+                  : [params.threshold.field],
+                value: params.threshold.value,
+                cardinality:
+                  params.threshold.cardinality != null ? params.threshold.cardinality : [],
+              }
+            : undefined,
+        timestampOverride: convertNullToUndefined(params.timestampOverride),
+        exceptionsList:
+          params.exceptionsList != null
+            ? params.exceptionsList
+            : params.exceptions_list != null
+            ? params.exceptions_list
+            : params.lists != null
+            ? params.lists
+            : [],
```

Comment:
@FrankHassanabad Is there any other conversion that needs to be done on these fields or is the rename enough?